internal/planner/plansdk/plansdk.go

```diff
@@ -7,16 +7,40 @@ import (
 	"fmt"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 	"time"
 
+	"github.com/samber/lo"
 	"go.jetpack.io/devbox/internal/envir"
+	"go.jetpack.io/devbox/internal/nix"
+	"go.jetpack.io/devbox/internal/searcher"
 )
 
 // FlakePlan contains the data to populate the top level flake.nix file
 // that builds the devbox environment
 type FlakePlan struct {
 	NixpkgsInfo *NixpkgsInfo
 	FlakeInputs []*FlakeInput
+
+	// Packages are used by "RemoveNixpkgs" feature
+	Packages []*Package
```

Comment:
I would really prefer you reuse `FlakeInputs`. This is basically the same. You can rename it to packages if you want (but that's a riskier change because you need to make sure you don't break templates.